### PR TITLE
issue #57: set default search source to azure

### LINKED ISSUE #57: Change Default Search Engine to Azure AI Search in finesse-frontend
Finesse-frontend currently uses AILab Search as its default search engine, which is undergoing development. To improve stability, switching to Azure AI Search is proposed. Also, to make future changes easier and avoid code alterations, an environment variable for selecting the default search engine should be implemented.

## Tasks
- [ ] Implement Environment Variable: Add an environment variable to select the default search engine.
- [ ] Configure Azure AI Search as Default: Set Azure AI Search as the default in the environment variable.
- [ ] Ensure Seamless Switching: Test to ensure that changing the search engine through the environment variable is effective.
- [ ] Implement Automated Tests: Create automated tests to confirm the functionality of the environment variable for different search engines, and to verify that Azure AI Search is the default when the variable is unset.
- [ ] Update Deployment Process: Ensure the new environment variable is included in the deployment pipeline.
- [ ] Update Project Documentation: Document the procedure for changing the default search engine using the environment variable.

## Acceptance Criteria
- Ability to select the default search engine in finesse-frontend via an environment variable.
- Azure AI Search set as the default engine upon initial setup.
- Functionality of switching search engines using the environment variable is verified.
- Automated tests confirm the reliability of the environment variable implementation and that Azure AI Search is the default when the environment variable is not set.
- The deployment process includes the new environment variable.
- Project documentation includes instructions for altering the default search engine with the environment variable.

### PR DESCRIPTION
**Pull request by k-allagbe**
Closes #57
- [x] create REACT_APP_SEARCH_SOURCE env var
- [x] set azure as default
- [x] manual tests
- [x] unit tests
- [x] instructions in .env.template
- [x] set env var in github action secrets
- [x] set env var in vercel

### CODE REVIEW

**Comment by k-allagbe** [comment]
@redat97 please, set the env var in vercel.

**Review by redat97 — APPROVED** [review]
update env variable in vercel, will create a workflow for that very soon though. Please check and lmk if there's any issue not working

**Comment by rngadam** [comment]
> update env variable in vercel, will create a workflow for that very soon though. Please check and lmk if there's any issue not working

the current deployed version still defaults to AI Lab search:

image

**Comment by k-allagbe** [comment]
> > update env variable in vercel, will create a workflow for that very soon though. Please check and lmk if there's any issue not working
> 
> the current deployed version still defaults to AI Lab search:
> 
> image

This is because that link points to the most recent vercel build which doesn't have the changes in this PR.
Use this link https://finesse-frontend-3oqzud7gu-reda-tarzalts-projects.vercel.app/ instead.

**Comment by redat97** [comment]
I changed the workflow to edit the comment and keep the preview URL to avoid this but I don't know if we want an URL with my name in it all over the repository
https://github.com/ai-cfia/finesse-frontend/pull/62

File: src/contexts/DataContext.tsx [hunk 1]
```diff
@@ -11,6 +11,20 @@ interface DataContextState {
   setQueryResult: React.Dispatch<React.SetStateAction<QueryResult[]>>;
 }
 
+// Utility function to map the environment variable to the SearchSource enum
+export const getDefaultSearchSource = (): SearchSource => {
```
Comment (rngadam) [inline src/contexts/DataContext.tsx:15]: I'm having trouble believing this is the simplest code for this when this works:

```typescript
const enum EDirection {
  Up,
  Down,
  Left,
  Right,
}

const lookup = EDirection["Down"]

console.log(lookup) // outputs 1
```

**Review by rngadam — CHANGES_REQUESTED** [review]

File: src/contexts/DataContext.tsx (hunk 1 above)
Comment (k-allagbe) [reply src/contexts/DataContext.tsx:15]: Lint does not allow the notation `EDirection["Up"]`. Says `["Up"] is better written in dot notation` meaning something like `EDirection.Up`. It pained me to commit it too. I searched for a better way with no avail. I didn't see the `EDirection["Up"]` notation before though.

Comment (k-allagbe) [reply src/contexts/DataContext.tsx]: Done. Strangely, `EDirection["Down"]` doesn't pass Lint. Says `["Down"] is better written in dot notation`. But `EDirection["Down" as keyof typeof Edirection]` passes.

**Comment by k-allagbe** [comment]
@redat97 these actions are not mine:
image

Is this you trying out your env solution? It's best if you create another branch for better tracking of who's doing what.

**Comment by redat97** [comment]
> @redat97 these actions are not mine: image
> 
> Is this you trying out your env solution? It's best if you create another branch for better tracking of who's doing what.
> 
> Also, we need to review carefully the action you're experimenting with `hackzcompany/vercel-env-push-action@v4`.

it's because it's your build that is says your name I'm not changing the code but yeah I did do it in another branch and it worked I applied it to your branch then had to make a small change so I ran it again and it had a weird error message and I ran it again to see if it was an actual issue or just a bug or something as that happens sometime I wasn't trying to debug it from your branch it just looks like I did it a lot of times because the notification appears for the on push event and pull request event

File: src/_versions.ts [hunk 2]
```diff
@@ -11,8 +11,8 @@ export interface TsAppVersion {
 export const versions: TsAppVersion = {
     version: '0.2.2',
```
Comment (rngadam) [inline src/_versions.ts:12]: this version should change? (from package.json)

**Review by rngadam — APPROVED** [review]
except for version that needs to be kept updated

File: src/_versions.ts (hunk 2 above)
Comment (k-allagbe) [reply src/_versions.ts]: Done

**Pull request merged**